File: kotlin/atbash-cipher/src/main/kotlin/AtbashCipher.kt
object Atbash {

    private fun tr(s: String, map: Map<Char, Char>) =
        s.toLowerCase().filter { it.isLetterOrDigit() }.map { if (it.isDigit()) it else map[it] }.joinToString("")

    fun encode(s: String) = tr(s, ('a'..'z').zip('z' downTo 'a').toMap())
        .chunked(5).joinToString(" ")

    fun decode(s: String) = tr(s, ('z' downTo 'a').zip('a'..'z').toMap())
}

File: ruby/atbash-cipher/atbash_cipher.rb
module Atbash
  ENC = ('a'..'z').to_a.zip(('a'..'z').to_a.reverse).to_h
  DEC = ('a'..'z').to_a.reverse.to_a.zip(('a'..'z').to_a).to_h

  def self.common(s, m)
    s.downcase.scan(/\w/).map { m[_1] || _1 }.join
  end

  def self.encode(s)
    common(s, ENC).scan(/.{1,5}/).join(" ")
  end

  def self.decode(s)
    common(s, DEC)
  end
end
